pkg/controller/docdb/dbclusterparametergroup/setup.go

```diff
@@ -0,0 +1,294 @@
+}
+
+func postObserve(_ context.Context, cr *svcapitypes.DBClusterParameterGroup, _ *svcsdk.DescribeDBClusterParameterGroupsOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return obs, err
+	}
+
+	cr.SetConditions(v1.Available())
+	return obs, err
+}
+
+func (e *hooks) lateInitialize(cr *svcapitypes.DBClusterParameterGroupParameters, resp *svcsdk.DescribeDBClusterParameterGroupsOutput) error {
+	group := resp.DBClusterParameterGroups[0]
+
+	paramsReq := &svcsdk.DescribeDBClusterParametersInput{DBClusterParameterGroupName: group.DBClusterParameterGroupName}
+	paramsResp, err := e.client.DescribeDBClusterParameters(paramsReq)
+	if err != nil {
+		return errors.Wrap(err, errDescribeParameters)
+	}
+
+	cr.CustomDBClusterParameterGroupParameters.Parameters = lateInitializeParameters(cr.Parameters, paramsResp.Parameters)
+	return nil
+}
+
+func lateInitializeParameters(in []*svcapitypes.Parameter, from []*svcsdk.Parameter) []*svcapitypes.Parameter {
+	out := in
+	if out == nil {
+		out = []*svcapitypes.Parameter{}
+	}
+
+	apiParams := make(map[string]*svcapitypes.Parameter, len(out))
+	for _, p := range out {
+		apiParams[awsclient.StringValue(p.ParameterName)] = p
+	}
+
+	for _, sdkP := range from {
+		if _, exists := apiParams[awsclient.StringValue(sdkP.ParameterName)]; !exists {
+			newP := &svcapitypes.Parameter{}
+			generateAPIParameter(sdkP, newP)
+			out = append(out, newP)
+		}
+	}
+
+	return out
+}
+
+func preUpdate(_ context.Context, cr *svcapitypes.DBClusterParameterGroup, obj *svcsdk.ModifyDBClusterParameterGroupInput) error {
+	obj.DBClusterParameterGroupName = awsclient.String(meta.GetExternalName(cr))
+	obj.Parameters = generateSdkParameters(cr.Spec.ForProvider.Parameters)
+	return nil
+}
+
+func (e *hooks) postUpdate(_ context.Context, cr *svcapitypes.DBClusterParameterGroup, resp *svcsdk.ModifyDBClusterParameterGroupOutput, upd managed.ExternalUpdate, err error) (managed.ExternalUpdate, error) {
+	if err != nil {
+		return upd, err
+	}
+	cr.Status.SetConditions(v1.Available())
+	return upd, svcutils.UpdateTagsForResource(e.client, cr.Spec.ForProvider.Tags, cr.Status.AtProvider.DBClusterParameterGroupARN)
+}
+
+func preCreate(_ context.Context, cr *svcapitypes.DBClusterParameterGroup, obj *svcsdk.CreateDBClusterParameterGroupInput) error {
+	obj.DBClusterParameterGroupName = awsclient.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func postCreate(_ context.Context, cr *svcapitypes.DBClusterParameterGroup, resp *svcsdk.CreateDBClusterParameterGroupOutput, cre managed.ExternalCreation, err error) (managed.ExternalCreation, error) {
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+
+	// CreateDBClusterParameterGroup does not create the parameters themselves. Parameters are added during update.
+	cr.Status.SetConditions(v1.Available())
+	meta.SetExternalName(cr, awsclient.StringValue(resp.DBClusterParameterGroup.DBClusterParameterGroupName))
```

Comment:
It seems we are able to use `awsclient.String(meta.GetExternalName(cr))` as the resource name for this resource. So, we need to add an [ignore statement](https://github.com/crossplane/provider-aws/blob/master/CODE_GENERATION.md#external-name) to exclude `DBClusterParameterGroupName` and just assume that the name comes from `awsclient.String(meta.GetExternalName(cr))`. You don't need to capture it in `postCreate`.

Reply:
Fixed